library/core/src/cmp.rs

```diff
@@ -2030,6 +2024,9 @@ mod impls {
         fn eq(&self, other: &&B) -> bool {
             PartialEq::eq(*self, *other)
         }
+        // if <A as PartialEq<B>>::ne uses inline assembly or FFI, then
+        // this forwarding impl may be more efficient than the default impl
+        #[allow(clippy::partialeq_ne_impl)]
```

Comment:
Probably `allow`->`expect` so we get a bit of noise if these change for some reason.

Maybe just make this a module-level attribute? Since it applies to all but one `fn ne` in this file.

Reply:
But at module level would it not also conflict with the few missing ne, at least as `expect`?

I was thinking of removing duplication a bit with a macro here, since the four cases of `(&mut, &mut)`, `(&mut, &)`, `(&, &mut)`, `(&, &)` are (nearly?) identical. That would also collapse most of these annotations...

Reply:
I've now collapsed this with macros to just one `PartialEq::ne` impl with an expect.